### PR TITLE
Add NTP client package

### PR DESCRIPTION
**Pull request by MarcFontaine**
This PR imports the pre-existing Ntp Client from `cardano-sl` into the repository.
We have decided to basically import the code AS IS (with only minimal changes)
and address the problems in the code step by step in separate PRs 
This PR addresses #1007

### CODE REVIEW

File: ntp/ntp.cabal [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name:                cardano-ntp
+name:                ntp
```
Comment (dcoutts) [inline ntp/ntp.cabal]: For later (not this PR): lets consider the package name, e.g. `ntp-client`.

File: ntp/ntp.cabal [hunk 2]
```diff
@@ -40,6 +35,28 @@ Library
                         OverloadedStrings
                         MonadFailDesugaring
 
+Executable ntp-app
```
Comment (dcoutts) [inline ntp/ntp.cabal]: I presume this is for experimenting during development and not something that's permanent.

Comment (coot) [reply ntp/ntp.cabal]: or `ntp-client-demo`

**Comment by coot** [comment]
I don't see tests in this `PR`; There were no tests in `cardano-sl`? or will they come in another PR?

**Pull request merged**